Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -149,10 +155,13 @@ void OpenALStream::SoundLoop()
   memset(uiBuffers, 0, numBuffers * sizeof(ALuint));
   uiSource = 0;
 
-  // Checks if a X-Fi is being used. If it is, disable FLOAT32 support as this sound card has no
-  // support for it even though it reports it does.
+  if (alIsExtensionPresent("AL_EXT_float32"))
+    float32_capable = true;
+
+  // Checks if a X-Fi is being used. If it is, enable FIXED32 support as this sound card
+  // is the only one that supports it for now.
   if (strstr(alGetString(AL_RENDERER), "X-Fi"))
-    float32_capable = false;
+    fixed32_capable = true;
```

Comment:
Yes, you are right. Anything after X-Fi (and some cards branded as X-Fi) from Creative uses only Sensaura's OpenAL driver.

Just tried 3DSoundBack. It only has a 32-bit OpenAL driver so it won't work with Dolphin.

Sadly RightMark is not enough, but you can at least test if there is any 64-bit support by going to your system32 folder and looking for a "cm_oal.dll" or any other file ending with "_oal.dll" which is not "soft_oal.dll" and "wrap_oal.dll" (if you are on a 64-bit Windows).